openmdao/core/problem.py

```diff
@@ -345,6 +351,18 @@ def check_partials(self, out_stream=sys.stdout, comps=None, compact_print=False,
 
         model = self.model
 
+        # Log to stdout on info channel if no logger is passed in.
+        if logger is None:
+            global _set_logger
+            if _set_logger is None:
+                logger = logging.getLogger("check_total_derivatives")
+                _set_logger = logger
+                console = logging.StreamHandler(sys.stdout)
+                console.setLevel(logging.INFO)
+                logger.addHandler(console)
+            else:
+                logger = _set_logger
+
```

Comment:
Actually, not quite duplicated, except I forgot to change the logger name when I copied it. I don't mind the duplication, but if we remove it, we also need to remove the code in check_config. That code is slightly different in that it specifies a custom format for the messages that are logged. Perhaps we need a new story for a unified logging manager.